internal/ctrlutils/ingress-status.go

```diff
@@ -52,6 +52,48 @@ type statusInfo struct {
 	hostname string
 }
 
+//  StatusUpdater is a Kubebuilder controller that updates resource statuses
+type StatusUpdater interface {
+	// NeedLeaderElection indicates if the proxy must only be run when it is controlled by a leader controller
+	NeedLeaderElection() bool
+
+	// Start starts the proxy update loop
+	Start(ctx context.Context) error
+
+	// PullConfigUpdate starts the config update routine
+	PullConfigUpdate(ctx context.Context)
+}
```

Comment:
Tried to do this originally returning a non-pointer and thought:

```
ctrlutils.StandardStatusUpdater does not implement manager.Runnable (Start method has pointer receiver)
```
was because it wasn't returning an interface (since that's what NewCacheBasedProxyWithStagger does). TIL it's actually whether it's a pointer or not, and the interface version has to be a pointer.

41c61b31